src/gui/editors/AutomationEditor.cpp

```diff
@@ -1308,12 +1308,26 @@ void AutomationEditor::paintEvent(QPaintEvent * pe )
 				}
 
 				float *values = m_pattern->valuesAfter( it.key() );
-				for( int i = 0; i < (it+1).key() - it.key(); i++ )
-				{
 
-					drawLevelTick( p, it.key() + i, values[i],
-									is_selected );
+				float nextValue;
+				if ( m_pattern->valuesAfter( ( it + 1 ).key() ) != NULL )
```

Comment:
You should have brackets for the if/else statement, and there should be no space between `if` and the parenthesis: `if(...`